src/Animations/babylon.runtimeAnimation.ts

```diff
@@ -495,30 +512,30 @@
                         break;
                     // Quaternion
                     case Animation.ANIMATIONTYPE_QUATERNION:
-                        offsetValue = new Quaternion(0, 0, 0, 0);
+                        offsetValue = _staticOffsetValueQuaternion;

```

Comment:
Can we create them on demand?

Reply:
It is generated at the top of the file and shared with every animation, it is a private const available to this file only

Reply:
Oh I missed that :)
Then you need to reset then to default value

Reply:
It is not changed, I tracked that value and it is read-only, do you still want it to be reset?

Reply:
Or what do you think about a Object.freeze? or extending the class with 
```ts
class StaticVector3 extends Vector3 { 
  get x() { return this.initialX } 
  set x(x) { throw }
  ...
}
```?

Reply:
We need to reset to default because it is used by the interpolator here:
https://github.com/BabylonJS/Babylon.js/blob/master/src/Animations/babylon.animation.ts#L815

Reply:
Ok, I'll do it, but those values are read-only. They are used like this

`return quatValue.addInPlace(offsetValue.scale(repeatCount));`
or
`return offsetValue * repeatCount + floatValue;`

And in every case, `T.scale(number)` returns a new `T` without mutating the original

Reply:
This is a really good point :)
So don't change them !

LGTM!